components/identity/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -354,21 +358,26 @@ public boolean isRefreshTokenRenewalEnabled() {
             synchronized (this) {
                 if (supportedClientAuthHandlers == null) {
                     supportedClientAuthHandlers = new ArrayList<ClientAuthenticationHandler>();
-                    for (String entry : supportedClientAuthHandlerClassNames) {
+
+                    for (Map.Entry<String, Properties> entry : supportedClientAuthHandlerData.entrySet()) {
                         ClientAuthenticationHandler clientAuthenticationHandler = null;
-                        try {
-                            clientAuthenticationHandler = (ClientAuthenticationHandler) Class.forName(entry).newInstance();
-                            clientAuthenticationHandler.init();
-                        } catch (InstantiationException e) {
-                            log.error("Error instantiating " + entry);
-                        } catch (IllegalAccessException e) {
-                            log.error("Illegal access to " + entry);
-                        } catch (ClassNotFoundException e) {
-                            log.error("Cannot find class: " + entry);
-                        } catch (IdentityOAuth2Exception e) {
-                            log.error("Error while initializing " + entry);
-                        }
-                        supportedClientAuthHandlers.add(clientAuthenticationHandler);
+                            try {
+                                clientAuthenticationHandler = (ClientAuthenticationHandler)
+                                        Class.forName(entry.getKey()).newInstance();
+                                clientAuthenticationHandler.init(entry.getValue());
+                                supportedClientAuthHandlers.add(clientAuthenticationHandler);
+
+                            //Exceptions necessarily don't have to break the flow since there are cases
+                            //runnable without client auth handlers
+                            } catch (InstantiationException e) {
+                                log.error("Error instantiating " + entry);
+                            } catch (IllegalAccessException e) {
+                                log.error("Illegal access to " + entry);
+                            } catch (ClassNotFoundException e) {
+                                log.error("Cannot find class: " + entry);
+                            } catch (IdentityOAuth2Exception e) {
+                                log.error("Error while initializing " + entry);
+                            }

```

Comment:
fixed in https://github.com/wso2/carbon-identity/pull/375/files#diff-7d4a030fe07f4cd074ef718f7a1a2d57R373